docs/pipeline-resources.md

```diff
@@ -0,0 +1,79 @@
+### Pipeline Resources
+
+## Git Resource
+
+Git resource represents a [git](https://git-scm.com/) repository, that containes the source code to be built by the pipeline. Adding the git resource as an input to a task will clone this repository and allow the task to perform the required actions on the contents of the repo.  
+
+Use the following example to understand the syntax and strucutre of a Git Resource
+
+ #### Create a git resource using the `PipelineResource` CRD
+ 
+ ```
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: Resource
+metadata:
+  name: wizzbang-git
+  namespace: default
+spec:
+  type: git
+  params:
+  - name: url
+    value: github.com/wizzbangcorp/wizzbang
+ ```
+
+   Params that can be added are the following:
+
+   1. URL: represents the location of the git repository 
+   1. Revision: Git [revision](https://git-scm.com/docs/gitrevisions#_specifying_revisions ) (branch, tag, commit SHA or ref) to clone. If no revision is specified, the resource will default to `latest` from `master`
+   1. Service Account: specifies the `name` of a `ServiceAccount` resource object. Add this paramater to run your task with the privileges of the specified service account. If no serviceAccountName field is specified, your task runs using the default service account that is in the namespace of the Pipeline resource object.  
+
+ #### Use the defined git resource in a `Task` definition
+
+```
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: Task
+metadata:
+  name: build-push-task
+  namespace: default
+spec:
+    inputs:
+        resources:
+           - name: wizzbang-git
+             type: git
```

Comment:
`type` is redundant here.  We can get the type for this Resource from the resource definition.

Reply:
we added `type` to make the task yaml more readable, I'd like to keep it at least for now

Reply:
ok sounds good.